src/cdk/overlay/position/cover-position-strategy-factory.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Inject, Injectable} from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+import {Platform} from '@angular/cdk/platform';
+import {ViewportRuler} from '@angular/cdk/scrolling';
+import {FlexibleConnectedPositionStrategyOrigin} from './flexible-positioning';
+import {CoverPositionStrategy} from './cover-position-strategy';
+
+@Injectable({providedIn: 'root'})
```

Comment:
Add JsDoc description

I'd also add a file-level comment that explains that this class exists so that people can instantiate a `CoverPositionStrategy` without manually passing in its numerous injected deps

Reply:
Will do - IMO all of the factories for position strategies should move to something like this rather than baking them into Overlay (and defeating tree-shaking of them in the process).